### PR TITLE
Switch to GitHub actions for better cross platform testing

### PR DESCRIPTION
**Pull request by dims**
Sets us up in a good position to add support for windows based testing later ( need https://github.com/kubernetes/utils/issues/153 to be fixed first) as we are facing issues as mentioned in https://github.com/kubernetes/utils/issues/142

### CODE REVIEW

**Comment by dims** [comment]
cc @neolit123

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Test
+on: [push, pull_request]
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-versions: [1.12.x, 1.13.x, 1.14.x]
+        platform: [ubuntu-latest, macos-latest]
```
Comment (neolit123) [inline .github/workflows/test.yml:8]: is windows-latest failing here too?
i'd enable it to know whether we need to log issue.

Comment (dims) [reply .github/workflows/test.yml:8]: @neolit123 i filed one already following your lead https://github.com/kubernetes/utils/issues/153

Comment (neolit123) [reply .github/workflows/test.yml:8]: ack, thanks

**Pull request merged**